From 615f852746f6a8c1331eec32e8352d7cfcd63f76 Mon Sep 17 00:00:00 2001
From: saiarcot895 <saiarcot895@gmail.com>
Date: Mon, 19 Dec 2016 10:35:14 -0800
Subject: [PATCH] Fix GN bootstrap.

Update list of required files and compile flags.

BUG=None

Review-Url: https://codereview.chromium.org/2588003002
Cr-Commit-Position: refs/heads/master@{#439509}
---
 tools/gn/bootstrap/bootstrap.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/tools/gn/bootstrap/bootstrap.py b/tools/gn/bootstrap/bootstrap.py
index bfdf0fc1ce64..c3642e8f487c 100755
--- a/tools/gn/bootstrap/bootstrap.py
+++ b/tools/gn/bootstrap/bootstrap.py
@@ -162,7 +162,10 @@ def build_gn_with_ninja_manually(tempdir, options):
       {'USE_EXPERIMENTAL_ALLOCATOR_SHIM': 'true' if is_linux else 'false'})
 
   write_buildflag_header_manually(root_gen_dir, 'base/debug/debugging_flags.h',
-      {'ENABLE_PROFILING': 'false'})
+      {
+          'ENABLE_PROFILING': 'false',
+          'ENABLE_MEMORY_TASK_PROFILER': 'false'
+      })
 
   if is_mac:
     # //base/build_time.cc needs base/generated_build_date.h,
@@ -402,6 +405,7 @@ def write_gn_ninja(path, root_gen_dir, options):
       'base/memory/ref_counted.cc',
       'base/memory/ref_counted_memory.cc',
       'base/memory/singleton.cc',
+      'base/memory/shared_memory_helper.cc',
       'base/memory/weak_ptr.cc',
       'base/message_loop/incoming_task_queue.cc',
       'base/message_loop/message_loop.cc',
@@ -456,6 +460,7 @@ def write_gn_ninja(path, root_gen_dir, options):
       'base/task_scheduler/scheduler_worker_pool_impl.cc',
       'base/task_scheduler/scheduler_worker_pool_params.cc',
       'base/task_scheduler/scheduler_worker_stack.cc',
+      'base/task_scheduler/scoped_set_task_priority_for_current_thread.cc',
       'base/task_scheduler/sequence.cc',
       'base/task_scheduler/sequence_sort_key.cc',
       'base/task_scheduler/task.cc',
-- 
2.11.0

